llvm/include/llvm/SandboxIR/SandboxIR.h

```diff
@@ -1954,6 +1962,55 @@ class Instruction : public sandboxir::User {
   }
   bool isShift() const { return cast<llvm::Instruction>(Val)->isShift(); }
   bool isCast() const { return cast<llvm::Instruction>(Val)->isCast(); }
+  bool isFuncletPad() const { return cast<llvm::Instruction>(Val)->isFuncletPad(); }
+  bool isSpecialTerminator() const { return cast<llvm::Instruction>(Val)->isSpecialTerminator(); }
+  bool isOnlyUserOfAnyOperand() const { return cast<llvm::Instruction>(Val)->isOnlyUserOfAnyOperand(); }
+  bool isLogicalShift() const { return cast<llvm::Instruction>(Val)->isLogicalShift(); }
+
+  //===--------------------------------------------------------------------===//
+  // Metadata manipulation.
+  //===--------------------------------------------------------------------===//
+
+  /// Return true if the instruction has any metadata attached to it.
+  bool hasMetadata() const { return cast<llvm::Instruction>(Val)->hasMetadata(); }
```

Comment:
Not sure about whether we should give access to metadata at this point, because these are LLVM IR structures and we can't really track their state.

Reply:
Removed getMetadata and getAllMetadata functions.